stargate-starter/src/main/java/io/stargate/starter/Starter.java

```diff
@@ -488,6 +488,17 @@ public void start() throws BundleException {
         }
 
         if (name.endsWith(".jar") && !name.startsWith("stargate-starter-")) {
+          // Any rate-limiting jar needs to be installed before the persistence-api is activated
+          // (because it needs to find the RateLimitingManager service), so we put it in front.
+          // But as we except the actual persistence implementation jar to be the very first, we
+          // add it as 2nd.
```

Comment:
I hope this should not be required when `DbActivator` does not declare a dependency on a `RateLimitingManager` in case it is not configured. OSGi plus `BaseActivator` should be able to resolve dependencies regardless of the order of bundle jars.

For reference: the persistence jar is put first in the list only to simplify the finding of the NodeTool class (as some other bundles may be able to load it too).

Reply:
Indeed. I added in earlier iterations when it wasn't working and I was figuring it out. But I removed it now and it does seem that it works fine.